design-documents/cart-discount-itemization.md

```diff
@@ -0,0 +1,58 @@
+**Cart Discount Itemization**
+
+Overview
+
+The individual itemized break down of the discounts applied at cart and cart item level are to be computed and persisted on every order.
+
+**Existing Implementation**
+
+The itemized discounts are computed and stored with extension attributes on 
+Magento\Quote\Api\Data\CartItemInterface 
+Magento\Quote\Api\Data\AddressInterface. 
+These are served via GraphQL cart query. The eventual end goal is to persist this data on every order.
+
+**Schema changes**
+
+The existing discount related metadata are being stored in quote_address, quote_item, sales_order, sales_order_item. But these tables tend to get large and have millions of records,
+and is decided its not ideal to alter heavyweight tables. So new tables have to be created to hold discounts data.
+
+Also have to be congizant about avoiding referencing between sales and quote tables.
+
+The data will be stored as serialized json to support extensibility.
+
+*quote_totals* 
```

Comment:
Please pick appropriate words for table name and descriptions. The goal is to save `discounts`, but the table name and descriptions are about `totals`.
Moreover, new tables introduce schemaless storage for read-only attributes without filtering capabilities. Thus, probably it's better to give a very common name for the table and use it for all types of read-only data(not only discounts). In this case, we need to create appropriate API and describe it in document

Reply:
Updated the doc with common table names.